### ISSUE TITLE
Clarification required about universal binaries for formula libmaxminddb

### ISSUE THREAD

**Issue by RR2DO2** [Homebrew/legacy-homebrew#37327]
Followup from the issue at their repository (https://github.com/maxmind/libmaxminddb/issues/82), the question is how universal binaries should be handled. By default it appears to install 64 bit only on 10.10.2, but universal is desired for Python interop.

**Comment by tdsmith** [comment]
Do you really want a universal libmaxminddb or do you just want your Python module to work with a universal python running in 64-bit mode? It's interesting that we don't run into this more often.

**Comment by RR2DO2** [comment]
Python on OS X is universal, since the library is 64 bit only, the python library that depends on it fails to compile on installation of the 32 bit version (https://github.com/maxmind/MaxMind-DB-Reader-python/issues/2). Since Python will not change (it's the OS installed version), the library will be built in universal, which means libmaxminddb needs to be universal too.

**Comment by tdsmith** [comment]
Well, an alternative is to do something like `if darwin and 64_bit: os.environ.setdefault("ARCHFLAGS", "-arch x86_64")` in setup.py to build a thin extension despite distutils's defaults. Homebrew learns whether a processor is 64-bit capable by examining `/usr/bin/sysctl hw.cpu64bit_capable`.

Calling `ENV.universal_binary` unconditionally in `install`, removing `option :universal`, and changing `depends_on`s to the form `depends_on "something" => ["universal", :optional]` would be sufficient to always build libmaxminddb as a universal binary. It's nice that there aren't m/any dependencies so that option isn't too annoying.

**Comment by mikemcquaid** [comment]
@tdsmith I'm :-1: on defaulting to universal being always on but depending on it being universal seems OK.

**Comment by tdsmith** [comment]
Sounds reasonable; if a universal python module is really desired maybe it's useful to add the python module to homebrew-python and set it up with a dependency on universal libmaxminddb.

**Comment by RR2DO2** [comment]
First question, why is universal not default?

So the Python module doesn't actually depend on any Homebrew formula at all, it just looks for the development headers installed as part of the libmaxminddb package.

While you could modify the Python package installation to force the architecture to be 64 bit only, this is not useful as Python itself is universal; so there is a valid reason of installing a universal version of the Python maxmind package. This then has a dependency on a universal libmaxminddb, where the Homebrew installation is the preferred way to install on OS X.

In light of all of this, I'm not sure why the libmaxminddb package wouldn't be universal? Unless you wouldn't use it with Python I guess. If the advised practise is to not force universal by default through Homebrew then maybe the Python package installation should have this as an installation note? However, this makes me wonder for how many other Python packages this is an issue - surely there are more than have a native component that have a dependency on a Homebrew package.

Or maybe the installation is flawed altogether and the Python installation package shouldn't rely on a Homebrew at all (albeit indirectly) and as part of its installation compile its own libmaxminddb locally.

Testing this with some other packages (for example pillow) they will succeed at install time, but fail at runtime if the 32 bit version of Python is used. In that case it fails to load libjpeg. So this is definitely a much wider issue.

```
$ arch -i386 python
Python 2.7.6 (default, Sep  9 2014, 15:04:36) 
[GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.39)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from PIL import Image
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "~/.virtualenvs/pillowtest/lib/python2.7/site-packages/PIL/Image.py", line 63, in <module>
    from PIL import _imaging as core
ImportError: dlopen(~/.virtualenvs/pillowtest/lib/python2.7/site-packages/PIL/_imaging.so, 2): Symbol not found: _jpeg_resync_to_restart
  Referenced from: ~/.virtualenvs/pillowtest/lib/python2.7/site-packages/PIL/_imaging.so
  Expected in: flat namespace
 in ~/.virtualenvs/pillowtest/lib/python2.7/site-packages/PIL/_imaging.so
>>> 
```

In short: either it seems the Homebrew formula should be universal, or the Python packages should not rely on the Homebrew installed version at all to be installed. The latter doesn't seem like a doable option as it is not the standard practise and would affect a very large amount of existing Python packages.

**Comment by tdsmith** [comment]
Universal builds aren't the default because they take longer and make binaries larger and aren't useful to most users; why build code you won't run for an architecture you don't need?

> Testing this with some other packages (for example pillow) they will succeed at install time, but fail at runtime if the 32 bit version of Python is used. In that case it fails to load libjpeg. So this is definitely a much wider issue.

I agree that it doesn't work but I don't understand why you'd do it. In the absence of a real-world use case I'm not worried about building things so that they play nicely with system Python in 32-bit mode without user intervention.

Methods of distributing Python extension modules are probably off-topic here but the latter approach is not uncommon and can make installation much simpler for users; pillow and scipy, for example, use [delocate](https://pypi.python.org/pypi/delocate) to bundle (universal) dylibs into wheels, which are served to most OS X users.

It's unusual for a Python extension module to fail to build in this situation. If decisions about how to build the Python module are made when the C library on which it depends is installed (i.e. through a configuration header), perhaps that's creating this failure. I don't expect that Homebrew's default behavior will change.

**Comment by RR2DO2** [comment]
I can see how this make sense from a lean installation POV. I'd argue that since 10.10 still supports 32 bit that there are entirely valid cases for universal binaries (otherwise Apple wouldn't still ship them). 

In light of the Homebrew ecosystem decisions I think the best next step is to actually modify the Python package for MaxMind to not rely on a universal binary, but can handle with 64-bit only being available on a mixed 32/64 bit system.

**Comment by tdsmith** [comment]
We'll accept a pull request if anything is actionable. Thanks!
